base58/src/lib.rs

```diff
@@ -4,6 +4,7 @@
 //!
 //! This crate can be used in a no-std environment but requires an allocator.
 
+#![cfg(feature = "alloc")]
```

Comment:
:open_mouth: does it actually work?

Anyway, I think I prefer a clear compile error with improvements to the CI rather than using black magic to make crate empty without explanation.

Reply:
I spent a bit of time playing with the various different ways to achieve this. I don't think this is black magic, it says "if alloc is not enabled this crate is empty" - which is a pretty good technical mapping from our "lets just move stuff from the always-alloc bitcoin crate to a new crate and later make the crate no-alloc"

Reply:
Well it's still easier to understand a compile message about missing alloc feature specifically than about missing items. Although IIRC there is WIP to have the compiler suggest turning on flags to get missing items.